### PR TITLE
Pull Request Template: add reviewer checklist

### PR DESCRIPTION
**Pull request by joe-kimmel-vmw**
#### What this PR does / why we need it:

This pull requests edits this template to add a checklist, which we said at retro might be helpful.

#### Which issue(s) this PR fixes:

action item from retro

#### Does this PR introduce a user-facing change?
nah

#### Additional Notes for your reviewer:

There would be a checklist here!

I think good questions for reviewing this change include:

- [ ] do we really want a checklist here?
- [ ] would you remove anything from the checklist in this PR?
- [ ] would you add anything to the checklist in this PR?

#### Additional documentation e.g., Proposal, usage docs, etc.:
this pull request is self-documenting ;-)

### CODE REVIEW

**Comment by cppforlife** [comment]
cc @danielhelfand @neil-hickey

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -24,6 +24,16 @@ Enter your extended release note in the block below.
 
 #### Additional Notes for your reviewer:
 
+##### Review Checklist:
+
+- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
+- [ ] Relevant tests are added or updated
+- [ ] Relevant internal docs added or updated
```
Comment (danielhelfand) [inline .github/PULL_REQUEST_TEMPLATE.md]: What is meant by internal here? If it's VMware specific, I would prefer to leave it out.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -24,6 +24,16 @@ Enter your extended release note in the block below.
 
 #### Additional Notes for your reviewer:
 
+##### Review Checklist:
```
Comment (danielhelfand) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: Can we add a bullet point for updating the kapp-controller dev.md doc if a change is development related?

Comment (joe-kimmel-vmw) [reply .github/PULL_REQUEST_TEMPLATE.md:27]: i think by internal (above) i meant "internal to this repo, such as dev.md" but your comments make it clear that meaning didn't come across :).  I'll update the wording!

**Review by danielhelfand — APPROVED** [review]
lgtm

Test failures seem like a flakey run

**Comment by neil-hickey** [comment]
> lgtm
> 
> Test failures seem like a flakey run

I went ahead and re-ran the things so hopefully next one will let us merge in :)

**Comment by danielhelfand** [comment]
> I went ahead and re-ran the things so hopefully next one will let us merge in :)

Did it fail again? That's the second time then as I also reran it.

**Pull request merged**